src/molecules/question/Question.tsx

```diff
@@ -446,6 +492,74 @@ export const QuestionData = ({
         />
       )
 
+    case UserQuestionType.File:
+      return (
+        <Controller
+          name={question.id}
+          control={control}
+          defaultValue={[]}
+          rules={{ required: config?.mandatory }}
+          render={({
+            field: { onChange: onControllerChange, onBlur, value },
+          }) => {
+            return (
+              <FileInputField
+                onChange={(files: Array<File>) => {
+                  void handleFilesUpload(files, onControllerChange)
+                }}
+                value={custom_json_parser(value as string)}
+                onBlur={onBlur}
+                accept={
+                  config?.file_storage?.accepted_file_types ?? [
+                    'application/pdf',
+                  ]
+                }
+                fileUploadErrors={fileUploadErrors}
+                onError={(error) => {
+                  setFileUploadErrors((prev) => [
+                    ...prev,
+                    { id: error, error: error as string },
+                  ])
+                }}
+              />
+            )
+          }}
+        />
+      )
+
+    case UserQuestionType.Image:
+      return (
+        <Controller
+          name={question.id}
+          control={control}
+          defaultValue={[]}
+          rules={{ required: config?.mandatory }}
+          render={({
+            field: { onChange: onControllerChange, onBlur, value },
+          }) => {
+            return (
+              <FileInputField
+                value={custom_json_parser(value as string)}
+                onChange={(files: Array<File>) => {
+                  void handleFilesUpload(files, onControllerChange)
+                }}
+                onBlur={onBlur}
+                accept={
+                  config?.file_storage?.accepted_file_types ?? ['image/*']
+                }
+                fileUploadErrors={fileUploadErrors}
+                onError={(error) => {
+                  setFileUploadErrors((prev) => [
+                    ...prev,
+                    { id: error, error: error as string },
+                  ])
+                }}
+              />
+            )
+          }}
+        />
+      )
+
```

Comment:
500+ lines of code, would be great to start splitting file and not extending it - not sure if this is a scope of this PR.

Reply:
Good suggestion 👍 
yes out-of-scope for now as it might divert the focus from delivering the new question type